contracts/modules/metadata/CoreMetadataViewModule.sol

```diff
@@ -0,0 +1,130 @@
+import { IPAccountStorageOps } from "../../lib/IPAccountStorageOps.sol";
+import { CORE_METADATA_VIEW_MODULE_KEY, CORE_METADATA_MODULE_KEY } from "../../lib/modules/Module.sol";
+import { ICoreMetadataViewModule, IViewModule } from "../../interfaces/modules/metadata/ICoreMetadataViewModule.sol";
+import { IModuleRegistry } from "contracts/interfaces/registries/IModuleRegistry.sol";
+
+/// @title Implementation of the ICoreMetadataViewModule interface
+/// @dev Provides functionalities to retrieve core metadata of IP assets, including name, description, and more.
+contract CoreMetadataViewModule is BaseModule, ICoreMetadataViewModule {
+    using IPAccountStorageOps for IIPAccount;
+
+    string public constant override name = CORE_METADATA_VIEW_MODULE_KEY;
+
+    address public immutable IP_ASSET_REGISTRY;
+    address public immutable MODULE_REGISTRY;
+
+    address public coreMetadataModule;
+
+    constructor(address ipAssetRegistry, address moduleRegistry) {
+        IP_ASSET_REGISTRY = ipAssetRegistry;
+        MODULE_REGISTRY = moduleRegistry;
+    }
+
+    /// @notice Updates the address of the CoreMetadataModule used by this view module.
+    /// @dev Retrieve the address of the CoreMetadataModule from the ModuleRegistry.
+    function updateCoreMetadataModule() external {
+        coreMetadataModule = IModuleRegistry(MODULE_REGISTRY).getModule(CORE_METADATA_MODULE_KEY);
+    }
+
+    /// @inheritdoc ICoreMetadataViewModule
+    function getName(address ipId) public view returns (string memory) {
+        string memory ipName = IIPAccount(payable(ipId)).getString(coreMetadataModule, "IP_NAME");
+        if (_isEmptyString(ipName)) {
+            ipName = IIPAccount(payable(ipId)).getString(IP_ASSET_REGISTRY, "NAME");
+        }
+        return ipName;
+    }
+
+    /// @inheritdoc ICoreMetadataViewModule
+    function getDescription(address ipId) public view returns (string memory) {
+        return IIPAccount(payable(ipId)).getString(coreMetadataModule, "IP_DESCRIPTION");
+    }
+
+    /// @inheritdoc ICoreMetadataViewModule
+    function getRegistrationDate(address ipId) public view returns (uint256) {
+        return IIPAccount(payable(ipId)).getUint256(IP_ASSET_REGISTRY, "REGISTRATION_DATE");
+    }
+
+    /// @inheritdoc ICoreMetadataViewModule
+    function getContentHash(address ipId) public view returns (bytes32) {
+        return IIPAccount(payable(ipId)).getBytes32(coreMetadataModule, "IP_CONTENT_HASH");
+    }
+
+    /// @inheritdoc ICoreMetadataViewModule
+    function getUri(address ipId) external view returns (string memory) {
+        return IIPAccount(payable(ipId)).getString(IP_ASSET_REGISTRY, "URI");
+    }
+
+    /// @inheritdoc ICoreMetadataViewModule
+    function getOwner(address ipId) public view returns (address) {
+        return IIPAccount(payable(ipId)).owner();
+    }
+
+    /// @inheritdoc ICoreMetadataViewModule
+    function tokenURI(address ipId) external view returns (string memory) {
```

Comment:
Is the IPAccount an ERC721? Naming this `token`Uri is confusing, since it's not part of a token contract.

Reply:
Agreed, It is mainly for suggest the returned JSON string according to NFT medatadata format. 
Renamed the function.